pkg/tcpip/transport/tcp/endpoint.go

```diff
@@ -933,6 +945,12 @@ func (e *endpoint) Readiness(mask waiter.EventMask) waiter.EventMask {
 		}
 	}
 	if e.EndpointState().connected() {
+		// Determine whether endpoint is half-closed with rcv shutdown
+		cs := e.connDirectionState()
+		if cs&connDirectionStateAll == connDirectionStateRcvClosed {
```

Comment:
These can be one line: `if cs := e.connDirectionState(); cs&connDirectionStateAll == connDirectionStateRcvClosed {`